static/app/views/alerts/rules/issue/setupMessagingIntegrationButton.tsx

```diff
@@ -27,77 +29,99 @@ function SetupMessagingIntegrationButton({projectSlug, refetchConfigs}: Props) {
   const organization = useOrganization();
 
   const onAddIntegration = () => {
-    refetch();
+    projectQuery.refetch();
     refetchConfigs();
   };
 
-  const {
-    data: project,
-    isLoading,
-    isError,
-    refetch,
-  } = useApiQuery<ProjectWithAlertIntegrationInfo>(
+  const projectQuery = useApiQuery<ProjectWithAlertIntegrationInfo>(
     [
       `/projects/${organization.slug}/${projectSlug}/`,
       {query: {expand: 'hasAlertIntegration'}},
     ],
     {staleTime: Infinity}
   );
 
-  const shouldRenderSetupButton = project && !project.hasAlertIntegrationInstalled;
+  const integrationQuery = useApiQuery<{providers: IntegrationProvider[]}>(
+    [
+      `/organizations/${organization.slug}/config/integrations/?provider_key=${providerKeys[0]}`,
```

Comment:
Can you add a comment here that checking only the first one is enough because they will all return the same response?
Just for code clarity, otherwise ideally in the future we should build things more like vertical by vertical vs integration by integration.